llvm/lib/Target/RISCV/RISCVRegisterInfo.cpp

```diff
@@ -585,6 +587,10 @@ bool RISCVRegisterInfo::eliminateFrameIndex(MachineBasicBlock::iterator II,
                (Lo12 & 0b11111) != 0) {
       // Prefetch instructions require the offset to be 32 byte aligned.
       MI.getOperand(FIOperandNum + 1).ChangeToImmediate(0);
+    } else if (Opc == RISCV::MIPS_PREFETCH && Subtarget.hasVendorXMIPSCBOP() &&
```

Comment:
You don't need to check `Subtarget.hasVendorXMIPSCBOP()`. If you see the opcode you can assume it must be enabled.